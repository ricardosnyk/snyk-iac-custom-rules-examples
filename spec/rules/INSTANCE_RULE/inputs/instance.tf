provider "aws" {
  region = "us-east-1"
}

resource "aws_instance" "valid" {
  ami                         = "ami-005e54dee72cc1d00" 
  instance_type               = "t2.micro"
  associate_public_ip_address = false

}

resource "aws_instance" "invalid" {
  ami                         = "ami-005e54dee72cc1d00" 
  instance_type               = "t2.micro"
  associate_public_ip_address = true
}