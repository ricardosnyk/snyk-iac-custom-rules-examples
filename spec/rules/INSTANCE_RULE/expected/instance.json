[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "aws_instance.invalid",
    "resource_namespace": "spec/rules/INSTANCE_RULE/inputs/instance.tf",
    "resource_type": "aws_instance",
    "severity": "critical",
    "resources": [
      {
        "id": "aws_instance.invalid",
        "type": "aws_instance",
        "namespace": "spec/rules/INSTANCE_RULE/inputs/instance.tf",
        "location": [
          {
            "filepath": "spec/rules/INSTANCE_RULE/inputs/instance.tf",
            "line": 12,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "associate_public_ip_address"
            ],
            "location": {
              "filepath": "spec/rules/INSTANCE_RULE/inputs/instance.tf",
              "line": 15,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "aws_instance.valid",
    "resource_namespace": "spec/rules/INSTANCE_RULE/inputs/instance.tf",
    "resource_type": "aws_instance",
    "severity": "critical",
    "resources": [
      {
        "id": "aws_instance.valid",
        "type": "aws_instance",
        "namespace": "spec/rules/INSTANCE_RULE/inputs/instance.tf",
        "location": [
          {
            "filepath": "spec/rules/INSTANCE_RULE/inputs/instance.tf",
            "line": 5,
            "column": 1
          }
        ]
      }
    ]
  }
]